dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/slicing/UsageSlicing.scala

```diff
@@ -370,7 +370,7 @@ object UsageSlicing {
   /** Adds extensions to extract all declaration nodes from a method body.
     */
   implicit class MethodDataSourceExt(trav: Iterator[Method]) {
-    def declaration: Iterator[Declaration] = trav.local ++ trav.parameter
+    def declaration: Iterator[Declaration] = trav.ast.collectAll[Declaration]
```

Comment:
This style of traversal using `declaration` is missing CALL nodes in python where some methods get imported and called directly without variable assignment.

Eg:

```python
from django.http import JsonResponse

def bar():
    return JsonReponse("foo")
```

Reply:
That is intentional, the usage slice is to describe usages of identifiers